### PR TITLE
ci: setup CI/CD

### PR DESCRIPTION
**Pull request by amopatel1234**
# Create Github workflows and other scripts 

Created workflow for raising pull requests, merging pull request and running a quality report for check code quality.
Scripts were adding to enable iOS to testes and built and deployed to app store connect

Note: these workflow and script still require setup work to fully function, these will be done at a later date

# Checklist

## Before raising your pull request:
- [x] Commit messages that conform to conventional commit messages
- [x] Ran the app locally ensuring it builds
- [ ] ~Ran the tests locally ensuring they pass on Build~
- [x] Pull request has a clear title with a short description about the feature or update
- [x] Created a `draft` pull request if it is not yet ready for review

## Before your pull request can be reviewed:
- [x] Met all of the acceptance criteria specified in the user story on Jira~
- [x] Reviewed your own code to ensure you are following the style guidelines
- [ ] ~Ran the app and tested the feature on a range of device sizes~
     ~Please include iPod Touch, iPhone SE and iPhone 11 as a minimum.~
- [ ] ~Written Unit and Integration tests if needed~

- [ ] ~Met all accessibility requirements?~
    - [ ] ~Checked dynamic type sizes are applied~
    - [ ] ~Checked VoiceOver can navigate your new code~
    - [ ] ~Checked a user can navigate only using a keyboard around your new code~

## Before merging your pull request:
- [x] Ensure that the code coverage and SonarCloud checks have passed
- [x] Actioned and resolved all comments, reaching out to reviewers for clarifications if necessary.
- [ ] ~Ran the app to ensure that no regressions have been caused by changes during code review.~

### CODE REVIEW

File: fastlane/Fastfile [hunk 1]
```diff
@@ -20,12 +20,12 @@ default_platform(:ios)
 platform :ios do
   desc "Run Tests and Output Code Coverage"
   lane :test do |options|
-    workspace = options[:workspaceFilename]
+    workspace = "OneLogin.xcworkspace"
```
Comment (amopatel1234) [inline fastlane/Fastfile:23]: Why was this changed? this value is passed in from the github action so if its wrong there we need change it on the action not here

File: fastlane/Fastfile [hunk 2]
```diff
@@ -20,12 +20,12 @@ default_platform(:ios)
...
 
     clear_derived_data
       
     run_tests(
-      workspace: workspace,
+      workspace: "OneLogin.xcworkspace",
```
Comment (amopatel1234) [inline fastlane/Fastfile:28]: same thing here, this could have been missed but need to pass this from the action so should be `options[:workspaceFilename]`

Comment (DylanRJ) [reply fastlane/Fastfile:28]: [Ibid](https://github.com/alphagov/di-mobile-ios-onelogin-app/pull/1#discussion_r1312772201)

**Review by DylanRJ — APPROVED** [review]

**Review by amopatel1234 — COMMENTED** [review]
Looks good to me, can be approved as I can not do it

**Review by mcgarj — APPROVED** [review]
Looks good to me!
